ArduCopter/GCS_Mavlink.cpp

```diff
@@ -6,10 +6,24 @@
 
 MAV_TYPE GCS_Copter::frame_type() const
 {
+    /*
+      for GCS don't give MAV_TYPE_GENERIC as the GCS would have no
+      information and won't display UIs such as flight mode
+      selection
+    */
+#if FRAME_CONFIG == HELI_FRAME
+    const MAV_TYPE mav_type_default = MAV_TYPE_HELICOPTER;
+#else
+    const MAV_TYPE mav_type_default = MAV_TYPE_QUADROTOR;
+#endif
     if (copter.motors == nullptr) {
-        return MAV_TYPE_GENERIC;
+        return mav_type_default;
+    }
+    MAV_TYPE mav_type = copter.motors->get_frame_mav_type();
+    if (mav_type == MAV_TYPE_GENERIC) {
```

Comment:
There is code in motors that returns generic by design for a miss configuration. I would prefer that we change that so it remains consistent across vehicles.

Reply:
the problem with generic is it gives the GCS no information on the set of flight modes
if we had a GENERIC_MULTIROTOR we could use that, but we don't

Reply:
What I mean is we should change it in motors not here. For example:

https://github.com/ArduPilot/ardupilot/blob/77e63da4b3a23ed988c6142ff99598cf6cf95b0b/libraries/AP_Motors/AP_MotorsMatrix.cpp#L1293-L1297

There is no point in having code to set the mavtype and then the only user of that code having more code to change it again. Might as well just change it in the original place.

Reply:
the point is the GCS API is specifically for the GCS, so changing it in just that one place won't have unforeseen consequences for future use cases